coursera/downloaders.py

```diff
@@ -61,6 +61,13 @@ def download(self, url, filename, resume=False):
 
         logging.debug('filename after get parameter removal: <%s>.', filename)
 
+        # FIXME (rbrito): This is pretty arbitrary and quite possibly wrong,
+        # as we don't even take extensions into account.
+        if len(filename) > 255:
+            filename = filename[:252] + '...'
```

Comment:
As for shortening, we also need to shorten each subdir because theoretically they may be long. I though of something like this but didn't have a chance to implement yet:
``` python
filename = os.path.join(map(shorten_filename, os.path.split(filename)))
```